### PR TITLE
#1959. Test valid combinations of modifiers. Part 1

### PR DESCRIPTION
**Pull request by sgrekhov**
It's only first half of the changes but, I believe, it's big enough for review. Some test case scipped because they are already tested in `basic_restrictions*` tests

### CODE REVIEW

**Review by eernstg — COMMENTED** [review]
Looks good!

I've added a bunch of comments, mostly typos.

I did suggest that we'd have a comment `// Reopen.` on every declaration that implies a reopening (and who knows whether I got that right ;-). That might not be very helpful, but I still think it's worth being aware of reopening as a mechanism.

File: LanguageFeatures/Class-modifiers/class_modifiers_lib.dart [hunk 1]
```diff
@@ -39,6 +39,20 @@ typedef TypedefAbstractBaseMixinClass = AbstractBaseMixinClass;
 typedef TypedefMixin = Mixin;
 typedef TypedefBaseMixin = BaseMixin;
 
+class ExtendsClass1 extends Class {}
+class ExtendsClass2 extends Class {}
+
+base class BaseExtendsBaseClass1 extends BaseClass {}
+base class BaseExtendsBaseClass2 extends BaseClass {}
+
+class ExtendsInterfaceClass1 extends InterfaceClass {}
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/class_modifiers_lib.dart]: Added

File: LanguageFeatures/Class-modifiers/class_modifiers_lib.dart [hunk 2]
```diff
@@ -39,6 +39,20 @@ typedef TypedefAbstractBaseMixinClass = AbstractBaseMixinClass;
...
+interface class ExtendsInterfaceClass2 extends InterfaceClass {}
+
+final class ExtendsFinalClass1 extends FinalClass {}
+base class ExtendsFinalClass2 extends FinalClass {}
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/class_modifiers_lib.dart]: Changed to `final` class to make exhaustiveness test more strong

File: LanguageFeatures/Class-modifiers/syntax_class_A01_t02.dart [hunk 3]
```diff
@@ -0,0 +1,92 @@
+// Copyright (c) 2023, the Dart project authors.  Please see the AUTHORS file
+// for details. All rights reserved. Use of this source code is governed by a
+// BSD-style license that can be found in the LICENSE file.
+
+/// @assertion Class with no modifiers can be constructed, extended, implemented
+/// but not mixed in and is not exhaustive
+///
+/// @description Checks that class with no modifiers can be constructed,
+/// extended and implemented in another library library
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/syntax_class_A01_t02.dart]: Fixed

File: LanguageFeatures/Class-modifiers/syntax_class_A02_t01.dart [hunk 4]
```diff
@@ -0,0 +1,125 @@
...
+//                                  ^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+base class BaseClassWithClass1 with Class {}
+//                                  ^^^^^
+// [analyzer] unspecified
+// [cfe] unspecified
+
+class BaseClassWithClass2 = Object with Class;
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/syntax_class_A02_t01.dart]: Strange. Fixed. Thank you!

File: LanguageFeatures/Class-modifiers/syntax_interface_class_A04_t01.dart [hunk 5]
```diff
@@ -0,0 +1,38 @@
...
+/// extended or mixed in and is not exhaustive
+///
+/// @description Checks that `interface class` is not exhaustive
+/// @author sgrekhov22@gmail.com
+
+// SharedOptions=--enable-experiment=class-modifiers
+
+import "class_modifiers_lib.dart";
+
+interface class C {}
```
[the remark this answers is not in the document]
Comment (sgrekhov) [reply LanguageFeatures/Class-modifiers/syntax_interface_class_A04_t01.dart]: Updated

**Review by eernstg — APPROVED** [review]
LGTM!

**Comment by sgrekhov** [comment]
@eernstg please merge this PR

**Comment by eernstg** [comment]
Ah, yes, I forgot: There is a syntax error in LanguageFeatures/Class-modifiers/syntax_sealed_class_A04_t01.dart, a mixin application class with a body. Fixed that. Landing.

**Comment by sgrekhov** [comment]
Thank you!

**Pull request merged**